### ISSUE TITLE
Proposing CIP-1: the Cartesi Improvement Proposal (CIP) system

### ISSUE THREAD

**Issue by fcecin** [cartesi-corp/cips#1]
I am a member of the Community, and I am proposing that the Cartesi Improvement Proposal (CIP) system, as described in the CIP-1 draft document (cip-1.md file in this repository), be the first governance process in Cartesi's governance system.

Link to the file: https://github.com/cartesi-corp/cips/blob/main/cips/cip-1.md

The CIP that I propose is heavily inspired in the EIP (Ethereum Improvement Proposal) system, with the main difference being this very Github issue: it represents the first post of a governance discussion thread that represents the formal beginning of the process of discussion of this system. This discussion can start with a simple "hey, why don't we do X?", or it can start with a mostly finished and consistent proposal, which is the case here.

Most proposals will not have the CIP-xxx.md file already drafted, let alone already committed to the repository. This is an exception. You can imagine that the cip-1.md file in the repository is "not really there." Here, in this issue, we will pretend that the file is not already there, that is, the CIP-1 does not exist yet as an accepted standard. And this discussion starts the beginning of the process (that's described in the CIP-1).

So, please, read the CIP-1, and comment here -- do you think it is a good idea? Do you have any objections, suggestions? This discussion is real, and the community -- for now, everyone in Cartesi -- has the power to change the CIP or even cancel its release through this meta-governance beta discussion of the CIP-1.

If this governance proposal passes, we will pilot with other test cases, other issues here, including ones that have pull requests for new CIP-xxx.md files, so we can iron out the details and properly pilot it.

**Comment by augustoteixeira** [comment]
I read it and I really like the proposal. I found the initial part of the process (the part about "filing an issue on the Cartesi Governance repository") to be a bit bureaucratic for the initial phase. But that is just my feeling.

**Comment by fcecin** [comment]
> I read it and I really like the proposal. I found the initial part of the process (the part about "filing an issue on the Cartesi Governance repository") to be a bit bureaucratic for the initial phase. But that is just my feeling.

The reason I want that step is to formalize the repository issue tracker as the place where all discussions about each CIP are either held, or are documented, curated.

When we were considering using a web forum, the requirement was posting the proposal as a forum post, to initiate the discussion. The only difference with respect to the EIP process is that in our case we are requiring the discussion to either be carried out here, or be referenced here (e.g. have a long forum thread elsewhere, and then file an issue that says: the first part of the discussion is here: <URL to the forum>).

When we discarded creating a dedicated web forum for governance discussions as the first use case of a Cartesi forum, we moved that function into the governance repository. In any case, we will have to give the community a forum, so I thought that we might as well make it formal that all discussions must be either held or be referenced, organized into one "Master" issue in the CIP repository that can then be referenced in the CIP document header itself (under 'discussions-to:').

This centralization of "official" CIP discussion in the governance repository also makes it easier to expand the CIP editor's power to include the  subjective evaluation of the maturity of the discussion itself. The discussion, by being held (or organized, referenced) in the governance repository itself, helps to make everyone aware that the maturity of the discussion is an object of governance that will be evaluated by the process -- the discussion data is "on-chain", as it were: the discussion and the standards are on the same database, because the decisions around the standards explicitly "read" (judge, filter) the length or age of the discussion object. Even better than having them on a dedicated Cartesi web forum, where other topics/rooms of discussion would also be included (we can also have that later, and those discussions can be referenced in the Master discussion issue that's a requirement to get a Draft standard accepted).

That power to judge the discussion maturity is one I gave to the CIP editors to protect the system from spam without requiring an actual developer that is not the editor to come in and provide a critique to each one of a potential array of spam submissions. So, if literally no developer says anything substantial about it, not even "looks OK", then the proposal cannot be accepted as a draft. I believe it is a good power that won't cost anything in terms of causing other problems. In contrast, the EIP editor seems to be just a pass-through for submitted documents, at least formally.

Of course, despite all this, we can simplify the process, by removing the requirement to post a master CIP issue on the repo, and the explicit power of the editor to ignore pull requests for CIP drafts that have an insufficient discussions-to. We can leave it all implicit, as it is in the EIP, and still, we can keep the Issue tracker of this repository as our supported space for governance discussions. And the community is then also free to use other forums elsewhere, and use links to those forums in the discussions-to: headers of their CIP draft submissions, exactly like the EIP.
